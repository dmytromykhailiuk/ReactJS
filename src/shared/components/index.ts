export * from "./Footer";
export * from "./Logo";
export * from "./Button";
export * from "./Input";
export * from "./ThreeDotsIcon";
export * from "./ErrorBoundary";
