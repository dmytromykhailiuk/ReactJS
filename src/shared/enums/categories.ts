export enum Categories {
  ALL = "ALL",
}
