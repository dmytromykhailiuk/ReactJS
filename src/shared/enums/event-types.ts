export enum EventTypes {
  CLICK = "click",
  SCROLL = "scroll",
}
