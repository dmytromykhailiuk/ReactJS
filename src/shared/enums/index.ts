export * from "./button-types";
export * from "./button-size";
export * from "./categories";
export * from "./sorting-options";
export * from "./movie-form-values";
export * from "./router-paths";
export * from "./modal-types";
