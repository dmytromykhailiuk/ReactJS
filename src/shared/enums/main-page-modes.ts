export enum MainPageModes {
  OVERVIEW = "overview",
  EDIT = "edit",
  CREATE = "create",
  DELETE = "delete",
}
