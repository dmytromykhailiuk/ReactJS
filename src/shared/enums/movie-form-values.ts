export enum MovieFormValues {
  MOVIE_ID = "id",
  TITLE = "title",
  RELEASE_DATE = "releaseDate",
  MOVIE_URL = "url",
  CATEGORY = "category",
  OVERVIEW = "overview",
  RUNTIME = "runtime",
  DURATION = "duration",
  RATING = "rating",
}
