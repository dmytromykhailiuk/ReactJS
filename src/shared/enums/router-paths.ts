export enum RouterPaths {
  HOME = '/',
  FILM = '/film/:id',
  SEARCH = '/search',
  ERROR = '/404',
}
