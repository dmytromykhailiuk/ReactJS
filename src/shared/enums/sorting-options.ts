export enum SortingOptionsProperties {
  RELEASE_DATE = "releaseDate",
  TITLE = "title",
}

export enum SortingOptions {
  releaseDate = "RELEASE DATE",
  title = "TITLE",
}
