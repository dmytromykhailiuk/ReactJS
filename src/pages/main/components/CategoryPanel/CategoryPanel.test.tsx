import { shallow } from "enzyme";
import React from "react";
import { CategoryPanel } from "./";
import toJson from 'enzyme-to-json';
import { Categories } from "../../../../shared/enums";

describe("CategoryPanel", () => {
  it("should match first snepshot", () => {
    const wrapper = shallow(
      <CategoryPanel
        selectedCategory={Categories.ALL}
        onChangeCategory={() => {}}
      />
    )
    expect(toJson(wrapper)).toMatchSnapshot("first");
  })

  it("should call 'onChangeCategory' if was click on category", () => {
    const onChangeCategory = jest.fn();
    const wrapper = shallow(
      <CategoryPanel
        selectedCategory={Categories.ALL}
        onChangeCategory={onChangeCategory}
      />
    );
    wrapper.find("li").last().simulate("click");
    expect(onChangeCategory).toHaveBeenCalled();
  })

  it("should call 'onChangeCategory' with 'Categories.ALL' if was click on first 'li'", () => {
    const onChangeCategory = jest.fn();
    const wrapper = shallow(
      <CategoryPanel
        selectedCategory={Categories.ALL}
        onChangeCategory={onChangeCategory}
      />
    );
    wrapper.find("li").first().simulate("click");
    expect(onChangeCategory).toHaveBeenCalledWith(Categories.ALL);
  })
});