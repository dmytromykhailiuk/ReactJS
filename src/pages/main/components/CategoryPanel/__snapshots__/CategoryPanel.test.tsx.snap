// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryPanel should match first snepshot: first 1`] = `
<ul
  className="category-panel"
>
  <li
    className="category-panel__item category-panel__item--active"
    onClick={[Function]}
  >
    ALL
  </li>
  <li
    className="category-panel__item"
    key="Documentary"
    onClick={[Function]}
  >
    Documentary
  </li>
  <li
    className="category-panel__item"
    key="Comedy"
    onClick={[Function]}
  >
    Comedy
  </li>
  <li
    className="category-panel__item"
    key="Horror"
    onClick={[Function]}
  >
    Horror
  </li>
  <li
    className="category-panel__item"
    key="Crime"
    onClick={[Function]}
  >
    Crime
  </li>
</ul>
`;
